auth/sasl/mech/plugins.go

```diff
@@ -0,0 +1,39 @@
+package mech
+
+import (
+	"fmt"
+	"sync"
+
+	log "github.com/golang/glog"
+)
+
+var (
+	mechLock       sync.Mutex
+	supportedMechs = make(map[string]Factory)
+)
+
+func Register(name string, f Factory) error {
+	mechLock.Lock()
+	defer mechLock.Unlock()
+
+	if _, found := supportedMechs[name]; found {
+		return fmt.Errorf("Mechanism registered twice: %s", name)
+	}
+	supportedMechs[name] = f
+	log.V(1).Infof("Registered mechanism %s", name)
+	return nil
+}
+
+func SelectSupported(mechanisms []string) (selectedMech string, factory Factory) {
```

Comment:
Ideally this would return a set of all matching names, and the client would then decide which one to use, then fetch it using a (currently non-existent) call like `GetFactory(selectedMech) (Factory, bool)`